ocaml/xapi-idl/lib_test/scheduler_test.ml

```diff
@@ -37,7 +37,13 @@ let timed_wait_callback ~msg ?(time_min = 0.) ?(eps = 0.1) ?(time_max = 60.) f =
         ()
       in
       f callback ;
-      let ready = Thread.wait_timed_read rd time_max in
+      let polly = Polly.create () in
+      let finally () = Polly.close polly in
+      Fun.protect ~finally @@ fun () ->
+      let ready =
+        Polly.wait polly 1 (time_max *. 1000. |> int_of_float) (fun _ _ _ -> ())
```

Comment:
We probably should give `Polly.wait` a named argument that includes the unit for the delay.

Reply:
This seems like a very complex way of doing Unix.sleepf, or I've accidentally dropped the file descriptor (again!).
And indeed I have, there should've been an `rd` here.

I'm defining a `wait_timed_read` in unixext to avoid this (and the other ) similar ubg.